React_Fundamentals/src/common/ToDoList/redux/components/TodoList.tsx

```diff
@@ -1,53 +1,74 @@
 import TodoItem from "./TodoItem";
 import { useSelector, useDispatch } from "react-redux";
-import { clearTodoList, getTodo } from "../actions";
+import { clearTodoList } from "../actions";
 import { TypeState as PropsList } from "../reducers/todos";
-import { useMemo } from "react";
-import { getTodoAPI } from "../../api/getTodoAPI";
+import { FormEvent, useEffect, useMemo, useState } from "react";
+import { fetchTodoList } from "../middlewares/fetchTodoList";
+import { Loading } from "./Loading";
+import { PropsList as PropsTodo} from "../reducers/todos";
 
-type todo = { id: number, text: string, completed: boolean }
-type stateType = {
-    todos: PropsList
+type rootState = {
```

Comment:
why not import `State` from `reducer/todos.ts` instead of creating another type?